lib/elixir/lib/registry.ex

```diff
@@ -171,6 +171,15 @@ defmodule Registry do
   @typedoc "The type of registry metadata values"
   @type meta_value :: term
 
+  @typedoc "A pattern to match on objects in a registry"
+  @type match_pattern :: atom | term
+
+  @typedoc "A guard to be evaluated when matching on objects in a registry"
+  @type guard :: {atom | term}
```

Comment:
This should be `tuple`. But we could be more detailed if tuple follows a specific format

Reply:
It is a tuple now. But the spec expresses a tuple of one element, and that is wrong

Reply:
I'll change it to tuple. I cannot find a type specification for guards in the Erlang documentation. What I've found is this informal grammar specification for the Match Specification which includes guards http://erlang.org/doc/apps/erts/match_spec.html and I think it's way to complex to be reduced to a type (at least for me) :)